### DISCUSSION TITLE
Adding new task into workflow

### DISCUSSION THREAD

**Discussion by mcgoldba** [ansys/pyfluent#2232] [Q&A]
I want to add the "Improve Surface Mesh" and "Improve Volume Mesh" tasks into the Watertight Geometry Workflow.  I suppose that this can be done with the `InsertNewTask()` method [here](https://fluent.docs.pyansys.com/version/stable/api/solver/datamodel/workflow/index.html#ansys.fluent.core.datamodel_232.workflow.Root.InsertNewTask), but it is unclear how this method should be called.  Can you provide an example to show how this should be done?

**Comment by DanielBajoCollados** [comment]
Sorry for the incredibly late answer.
The process would not use `InsertNewTask()`, but `InsertNextTask()` inside the TaskObject. That is:
`workflow.TaskObject["Generate the Surface Mesh"].InsertNextTask(CommandName="ImproveSurfaceMesh")`
